validator/src/validation/range.rs

```diff
@@ -1,67 +1,59 @@
-use crate::validation::Validator;
-
-/// Validates that a number is in the given range
+/// Validates that the given `value` is inside the defined range. The `max` and `min` parameters are
+/// optional and will only be validated if they are not `None`
 ///
-/// TODO: see if can be generic over the number type
 #[must_use]
-pub fn validate_range(range: Validator, val: f64) -> bool {
-    match range {
-        Validator::Range { min, max } => {
-            if let Some(m) = min {
-                if val < m {
-                    return false;
-                }
-            }
-
-            if let Some(m) = max {
-                if val > m {
-                    return false;
-                }
-            }
+pub fn validate_range<T>(value: T, min: Option<T>, max: Option<T>) -> bool
+where
+    T: PartialOrd + PartialEq,
+{
+    if let Some(max) = max {
+        if value > max {
+            return false;
+        }
+    }
 
-            true
+    if let Some(min) = min {
+        if value < min {
+            return false;
         }
-        _ => unreachable!(),
     }
+
+    true
 }
 
 #[cfg(test)]
 mod tests {
-    use super::{validate_range, Validator};
-
-    #[test]
-    fn test_validate_range_ok() {
-        let validator = Validator::Range { min: Some(0.0), max: Some(10.0) };
-        assert_eq!(validate_range(validator, 1_f64), true);
-    }
-
-    #[test]
-    fn test_validate_range_fail() {
-        let validator = Validator::Range { min: Some(0.0), max: Some(10.0) };
-        assert_eq!(validate_range(validator, 20_f64), false);
-    }
+    // This is needed to suppress the `validate_range` depreciation notice
+    // We want to keep the tests to make sure that this function will remain valid
```

Comment:
These comments are out of date they explained the `[allow(depeciated)]` from #132. These comments have also been removed in #134